### DISCUSSION TITLE
zh/guides/deepstream-nvidia-jetson/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#17651] [Docs]
# zh/guides/deepstream-nvidia-jetson/

了解如何使用TensorRT 和 DeepStream SDK 在NVIDIA Jetson 设备上部署Ultralytics YOLO11。探索性能基准，最大限度地提高人工智能能力。

https://docs.ultralytics.com/zh/guides/deepstream-nvidia-jetson/

**Comment by dao027** [comment]
do you have an implement of the bytetrack with deepstream?

**Reply by glenn-jocher** [reply]
@dao027 currently, there is no implementation of ByteTrack with DeepStream in our repository. However, you can explore [DeepStream-Yolo](https://github.com/marcoslucianops/DeepStream-Yolo) for YOLO integrations with DeepStream, which might be adapted for tracking solutions.

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Chinese Documentation](https://docs.ultralytics.com/zh/guides/deepstream-nvidia-jetson/) for guidance on using TensorRT and the DeepStream SDK for deploying Ultralytics YOLO11 on NVIDIA Jetson devices. This guide will help you explore performance benchmarks and enhance AI capabilities.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.

An Ultralytics engineer will assist you soon. If you have specific queries or need further details, sharing more information about your setup and any challenges you're facing will be helpful.

For further community support, join us on [Discord](https://ultralytics.com/discord) 🎧 for real-time conversations or participate in in-depth discussions on [Discourse](https://community.ultralytics.com). You can also engage with others on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you're using the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) by upgrading:

```bash
pip install -U ultralytics
```

## Environments

YOLO runs smoothly in these verified configurations (CUDA, CUDNN, Python, PyTorch preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. This ensures the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across multiple platforms. 

We're here to help! 😊

**Comment by lovechang1986** [comment]
I'm working on integrating SAHI with YOLOv11 and deploying it on a Jetson device using DeepStream. The original image size is 1440*1080, and I plan to split it into 640*640 sub-images with overlaps between different sub-images. Are there any good methods for this? 

Additionally, I still hope to use DeepSort for inference and tracking after obtaining the final image.

**Reply by glenn-jocher** [reply]
SAHI-style slicing isn’t something DeepStream handles automatically, so you’ll either need to slice frames before they enter the DeepStream pipeline (then offset + merge all tile detections back into full-frame coordinates *before* tracking), or implement ROI/tiling as a DeepStream preprocess step (e.g., via `nvdspreprocess` or a small custom plugin) so the primary GIE sees a batched set of crops; on Jetson it’s often simpler/faster to avoid slicing and instead run a single-pass TensorRT engine at a larger `imgsz` if latency allows (and YOLO26 is the recommended starting point for new deployments). If you plan to track inside DeepStream, put the tracker after you’ve merged detections to full-frame coordinates; DeepStream’s built-in `nvtracker` works out of the box, while DeepSORT specifically typically requires custom integration rather than a drop-in switch. If you paste your JetPack/DeepStream versions and your current `deepstream_app_config.txt` into this discussion, we can suggest the cleanest path based on the exact pipeline; the baseline export + DeepStream wiring is in the [Jetson DeepStream guide](https://docs.ultralytics.com/guides/deepstream-nvidia-jetson/) and [Jetson quickstart](https://docs.ultralytics.com/guides/nvidia-jetson/).

**Comment by Llinz24** [comment]
Is YOLO26-seg currently supported for direct deployment and execution in DeepStream? Are there any available tutorials or reference solutions?

**Reply by glenn-jocher** [reply]
Not in the published direct DeepStream flow yet—the current [Jetson + DeepStream guide](https://docs.ultralytics.com/guides/deepstream-nvidia-jetson/) is detection-focused and uses `DeepStream-Yolo`, while the available segmentation reference is [DeepStream-Yolo-Seg](https://github.com/marcoslucianops/DeepStream-Yolo-Seg), which currently lists `YOLO11-Seg` and `YOLOv8-Seg` support rather than `YOLO26-seg`, so there isn’t an official YOLO26-seg DeepStream tutorial at the moment; feel free to keep the details here in Discussions and we can help with the closest path forward. ([docs.ultralytics.com](https://docs.ultralytics.com/guides/deepstream-nvidia-jetson/))

**Reply by Llinz24** [reply]
I'm running into an issue with YOLO11-seg + DeepStream and could use some help 🙏
So I exported my YOLO11-seg model to ONNX and then to a TensorRT engine. When I run deepstream-app, it does find and load the engine just fine — but the segmentation part just doesn't work. I'm not seeing any masks or even bounding boxes on the video output. Any idea what might be going wrong here?
Also, when I use the deepstream-yolo export script to get the ONNX file, the output only shows one node:
1 OUTPUT kFLOAT output 100x102406
But from what I've seen with official YOLO-seg exports (like from Ultralytics), there should usually be two outputs — one for boxes/classes and another for the masks. That mismatch feels like it might be part of the problem?
Has anyone run into this before? Any tips on:
How to export the ONNX correctly for YOLO-seg (maybe a flag I'm missing?)
What to tweak in the DeepStream config (config_infer_primary.txt, parser settings, etc.) to make it actually show the segmentation results?
Thanks in advance!

**Reply by glenn-jocher** [reply]
Yes, a single-output ONNX is very likely the blocker here: for YOLO11-seg, DeepStream needs the segmentation export with both detection and mask-proto outputs, so a detection-style ONNX/parser will often load the engine but show neither boxes nor masks; please re-export from the Ultralytics side using the workflow in the [TensorRT export docs](https://docs.ultralytics.com/integrations/tensorrt/), make sure you are using the segmentation-specific DeepStream parser/config rather than the detection one, and if it still fails please paste the full `deepstream-app` log plus `yolo checks` output here so we can verify the tensor names, parser, and engine match.
